File: ruby/lib/trailer_vote/media_types/types/product_image_type.rb
# frozen_string_literal: true

require 'media_types'

module TrailerVote
  module MediaTypes
    module Types
      # noinspection RubyConstantNamingConvention
      ProductImageType = ::MediaTypes::Scheme.AnyOf(
        'backdrop',
        'poster'
      )

      # noinspection RubyConstantNamingConvention
      ProductImageTypeV1 = ::MediaTypes::Scheme.AnyOf(
        'backdrop',
        'poster',
        'Poster',
        'HeroMobileDynamic',
        'PosterDynamic',
        'HeroDesktopDynamic'
      )
    end
  end
end

File: typescript/application/types/product_image_type.ts
import { ElementType, literals } from '../utils/literals'

export const ProductImageTypesV1 = literals('backdrop', 'poster')
export type ProductImageTypeV1 = ElementType<typeof ProductImageTypesV1>

export const ProductImageTypesV0 = literals('backdrop', 'poster', 'Post', 'HeroMobileDynamic', 'PosterDynamic', 'HeroDesktopDynamic')
export type ProductImageTypeV0 = ElementType<typeof ProductImageTypesV0>

